lib/api/logging/index.js

```diff
@@ -92,9 +84,34 @@ if (isDevelopment && isConsoleAvailable) {
     }));
 }
 
+const logger = new winston.Logger({ transports });
+
+logger.addFileTransport = appPath => (
+    new Promise(resolve => {
+        const logDir = path.join(userDataDir, path.basename(appPath));
+        mkdirIfNotExists(logDir).then(() => {
+            logFilePath = path.join(logDir, `${filePrefix}-log.txt`);
+            const fileTransport = new winston.transports.File({
+                name: 'file',
+                filename: logFilePath,
+                level: 'debug',
+                json: false,
+                timestamp: () => new Date(),
+                formatter: logFormatter,
+            });
+            logger.add(fileTransport, {}, true);
+        })
+        .then(resolve)
```

Comment:
For my understanding; what function does this *then* have?

Reply:
The outer Promise needs to be resolved, taking this away will result in a blocked state.